cmd/kubeadm/app/phases/apiconfig/clusterroles.go

```diff
@@ -84,6 +85,10 @@ func CreateRBACRules(clientset *clientset.Clientset) error {
 	if err := createClusterRoleBindings(clientset); err != nil {
 		return err
 	}
+	if err := deletePermissiveNodesBindingWhenUsingNodeAuthorization(clientset); err != nil {
```

Comment:
Then the target version being deployed should be part of the input. I'd point out that tightening roles is generally something you would do *after* a deployment completed, because of version skew between components